libpromises/evalfunction.c

```diff
@@ -2066,6 +2066,144 @@ static FnCallResult FnCallClassify(EvalContext *ctx,
     return FnReturnContext(is_defined);
 }
 
+/*********************************************************************/
+
+// Calculates major.minor.patch version "value"
+static long VersionValue(char *const ver_string)
+{
+    assert(ver_string != NULL);
+
+    int mod = 10000;
+    char *endptr = NULL;
+    char *str_ptr = ver_string;
+    long res = 0;
+    for (int i = 0; i < 3; i++)
+    {
+        long ver = strtoul(str_ptr, &endptr, 10);
+        res += ver * mod;
```

Comment:
I don't think you should do it this way. Split the number in 3 (major, minor, patch). Then just compare major first, then minor, then patch.

Please take a look at `macro_if_version` in `cf3lex.l`, and try to align with how that works (maybe even make a shared function from the logic there).